ldap/servers/slapd/back-ldbm/ldbm_bind.c

```diff
@@ -63,13 +64,33 @@ ldbm_back_bind(Slapi_PBlock *pb)
      * find the target entry.  find_entry() takes care of referrals
      *   and sending errors if the entry does not exist.
      */
-    if ((e = find_entry(pb, be, addr, &txn, &result_sent)) == NULL) {
-        rc = SLAPI_BIND_FAIL;
-        /* In the failure case, the result is supposed to be sent in the backend. */
-        if (!result_sent) {
-            slapi_send_ldap_result(pb, LDAP_INAPPROPRIATE_AUTH, NULL, NULL, 0, NULL);
+    {
+        Slapi_DN *sdn = NULL;
+        const char *dn = NULL;
+        passwdPolicy *pwpolicy = NULL;
+
+        /* in case of lockout password policy we may need to update
+         * the target entry (when sending the result via update_pw_retry).
+         * In such case we need lock it in the cache
+         */
+        slapi_pblock_get(pb, SLAPI_TARGET_SDN, &sdn);
+        dn = slapi_sdn_get_dn(sdn);
+        pwpolicy = new_passwdPolicy(pb, dn);
+        if (pwpolicy && (pwpolicy->pw_lockout == 1)) {
```

Comment:
Doesn't this mean that these entries will be permanently pinned in the cache? That could cause the cache to thrash pretty hard in some setups.

Reply:
Yes, but my point is there is no condition to *unpin* an entry. This means if the cache has 4000 slots, there are 3500 users, then only 500 cache slots will be usable for anything else. This problem is worse if the number of users exceeds the cache size since then they won't be cached at all I would expect.

Reply:
Sorry but I do not understand your point:
  If I understand rightly this fix, the entry is pinned when getting it then unpinned (with cache_unlock_entry) once ldap_send_result (which may use the entry) is completed. So the maximum number of entries pinned by this fix is limited to the maximum number of worker thread (and not the number of users)